server/graphql/v1/CollectiveInterface.js

```diff
@@ -1533,6 +1543,10 @@ const CollectiveFields = () => {
         ) {
           paymentMethods = paymentMethods.filter(pm => !(pm.service === 'opencollective' && pm.type === 'collective'));
         }
+        // Remove payment methods that have expired
+        if (!args.includeRecentlyExpiredGiftCards) {
+          paymentMethods = paymentMethods.filter(pm => +new Date(pm.expiry) > +new Date());
```

Comment:
When `includeRecentlyExpiredGiftCards` is true, this will return **all** expired payment methods. We should only return expired **gift cards** (service='opencollective' and type='virtualcard'). We don't want to return expired credit cards.